### PR TITLE
[Java] Parquet reading example fails with Arrow v8.0

### PR DESCRIPTION
**Pull request by davisusanibar**
PR to fix problems mention at: https://github.com/apache/arrow-cookbook/issues/206

### CODE REVIEW

**Comment by davisusanibar** [comment]
Hi @lidavidm please could you help with a review.

File: java/source/dataset.rst [hunk 1]
```diff
@@ -224,11 +240,19 @@ Consider that we have these files: data1: 3 rows, data2: 3 rows and data3: 250 r
         scanner.scan().forEach(scanTask-> {
             VectorLoader loader = new VectorLoader(vsr);
             final int[] count = {1};
-            scanTask.execute().forEachRemaining(arrowRecordBatch -> {
-                loader.load(arrowRecordBatch);
-                System.out.println("Batch: " + count[0]++ + ", RowCount: " + vsr.getRowCount());
-                arrowRecordBatch.close();
-            });
+            try(ArrowReader reader = scanTask.execute()){
```
Comment (toddfarmer) [inline java/source/dataset.rst:243]:
```suggestion
            try (ArrowReader reader = scanTask.execute()) {
```

**Comment by toddfarmer** [comment]
Should we add any comments around these code changes to highlight that the supplied example code is version (8.0.0)-specific?

**Comment by lidavidm** [comment]
Maybe we should label the top level page, or perhaps include it in the page titles, or something?

Also, I wonder if we can just snapshot the 'old' cookbook on each release and label it as the '7.0.0 cookbook' or something

File: java/source/demo/pom.xml [hunk 2]
```diff
@@ -21,7 +21,7 @@
     <properties>
         <maven.compiler.source>8</maven.compiler.source>
         <maven.compiler.target>8</maven.compiler.target>
-        <arrow.version>7.0.0</arrow.version>
+        <arrow.version>8.0.0</arrow.version>
```
Comment (lidavidm) [inline java/source/demo/pom.xml:24]: We need to get this added to the list of post-release tasks and/or make it one of the things we check as part of release.

File: java/source/dataset.rst [hunk 3]
```diff
@@ -275,11 +302,19 @@ In case we need to project only certain columns we could configure ScanOptions w
...
-                System.out.print(vsr.contentToTSVString());
-                arrowRecordBatch.close();
-            });
+            try(ArrowReader reader = scanTask.execute()){
+                while (reader.loadNextBatch()) {
+                    try(VectorSchemaRoot root = reader.getVectorSchemaRoot()) {
+                        final VectorUnloader unloader = new VectorUnloader(root);
+                        try(ArrowRecordBatch arrowRecordBatch = unloader.getRecordBatch()){
+                            loader.load(arrowRecordBatch);
+                            System.out.print(vsr.contentToTSVString());
```
Comment (lidavidm) [inline java/source/dataset.rst:311]: nit: why not just `root.contentToTSVString()`? Is there a need to demonstrate VectorLoader/VectorUnloader here?

**Review by lidavidm — APPROVED** [review]

File: java/source/dataset.rst (hunk 1 above)
Comment (davisusanibar) [reply java/source/dataset.rst]: Thank you @toddfarmer, changed lots of places.

**Review by toddfarmer — APPROVED** [review]
LGTM!  Thanks @davisusanibar !

**Comment by davisusanibar** [comment]
> Maybe we should label the top level page, or perhaps include it in the page titles, or something?
> 
> Also, I wonder if we can just snapshot the 'old' cookbook on each release and label it as the '7.0.0 cookbook' or something

Ticket created: https://github.com/apache/arrow-cookbook/issues/208

File: java/source/dataset.rst (hunk 3 above)
Comment (davisusanibar) [reply java/source/dataset.rst]: Updated

**Comment by lidavidm** [comment]
Note we can't deploy cookbook updates until we get the R cookbook fixed (#201)

**Comment by davisusanibar** [comment]
What about [stackoverflow question](https://stackoverflow.com/questions/72233354/how-to-read-parquet-files-into-tables-in-java-using-apache-arrow)?

**Comment by lidavidm** [comment]
Do you want to just answer it?

**Comment by davisusanibar** [comment]
> Do you want to just answer it?

Yes, just mention the PR in case they need to continue with their work.

**Review by lidavidm — APPROVED** [review]

**Pull request merged**